src/main/java/io/cdap/e2e/pages/actions/CdfStudioActions.java

```diff
@@ -81,4 +83,35 @@ public static String bannerErrorMessage() {
   public static void clickSink() {
     CdfStudioLocators.sink.click();
   }
+
+  public static void clickValidateButton() {
+    CdfStudioLocators.validateButton.click();
+  }
+
+  public static void clickProperties(String plugin) {
+    SeleniumDriver.getDriver().findElement(
+      By.xpath("//*[contains(@title,'" + plugin + "')]//following-sibling::div")).click();
+  }
+
+  public static void  connectSourceAndSink(String source, String sink) {
+    SeleniumHelper.waitElementIsVisible(SeleniumDriver.getDriver().findElement(
+      By.xpath("//*[contains(@title,'" + sink + "')]")));
+    SeleniumHelper.dragAndDrop(
+      SeleniumDriver.getDriver().findElement(By.xpath("//*[contains(@class,'plugin-endpoint_" + source + "')]")),
+      SeleniumDriver.getDriver().findElement(By.xpath("//*[contains(@title,'" + sink + "')]")));
+  }
+
+  public static void selectBQ() throws InterruptedException {
```

Comment:
BQ should not belong to e2e. Only basic GCS actions should be here that will be reused across different plugins

Reply:
BigQuery is being used as Sink across different plugin testcases. thats why it is placed in e2e.
@ksahilCF

Reply:
Why? Do we have some test cases with non-google-cloud source and BQ sinks?

Reply:
Yes, we do have SAP data sources with BQ sink

Reply:
OK